test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/MSBuildAPIUtilityTests.cs

```diff
@@ -572,13 +571,50 @@ public async Task GetProjectsFromSolution_WithSolutionFile_ReturnsCorrectAbsolut
             Path.IsPathRooted(projectList.Last()).Should().BeTrue();
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GetProjectsFromSolution_WithSolutionFilter_ReturnsFilteredProjects(bool useSlnx)
+        {
+            // Arrange
+            var pathContext = new SimpleTestPathContext();
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot, useSlnx);
+            var tfm = FrameworkConstants.CommonFrameworks.NetStandard20;
+
+            var projectA = SimpleTestProjectContext.CreateNETCore("a", pathContext.SolutionRoot, tfm);
+            var projectB = SimpleTestProjectContext.CreateNETCore("b", pathContext.SolutionRoot, tfm);
+
+            solution.Projects.Add(projectA);
+            solution.Projects.Add(projectB);
+            solution.Create(pathContext.SolutionRoot);
+
+            var slnfContents = $$"""
+                {
+                    "solution": {
+                        "path": "solution.{{(useSlnx ? "slnx" : "sln")}}",
+                        "projects": [
+                            "a\\a.csproj"
+                        ]
+                    }
+                }
+                """;
+            var slnfPath = Path.Combine(pathContext.SolutionRoot, "filter.slnf");
+            File.WriteAllText(slnfPath, slnfContents);
+
+            // Act
+            var projectList = MSBuildAPIUtility.GetProjectsFromSolution(slnfPath);
+
+            // Assert
+            projectList.Should().BeEquivalentTo([projectA.ProjectPath]);
```

Comment:
Is there a reason to use BeEquivalentTo() here over something like Be()? Looking at https://fluentassertions.com/strings/ it seems the BeEquavlentTo supports case insensitivity and ignores order in a list.

Reply:
projectList is a collection, so this checks that the collection size is 1, in addition to the value of the single entry.

Reply:
Nice!